geode-benchmarks/src/main/java/org/apache/geode/benchmark/tests/PartitionedWithDeltaAndUniqueObjectReferenceBenchmark.java

```diff
@@ -0,0 +1,242 @@
+    config.warmupSeconds(30);
+    ClientServerTopology.configure(config);
+    config.before(new CreatePartitionedRegion(), SERVER);
+    config.before(new CreateClientPool(), CLIENT);
+    config.before(new CreateClientProxyRegionWithPool(), CLIENT);
+    config.workload(new PutWithDeltaTask(ID), CLIENT);
+    return config;
+  }
+
+  public static class PutWithDeltaTask extends BenchmarkDriverAdapter implements Serializable {
+
+    private Region<String, Session> region;
+    private final AtomicReference<String> sessionId = new AtomicReference<>(null);
+
+    private final String id;
+
+    private final List<String> existingSessionAttributeNames =
+        new ArrayList<>();
+
+    private final Random random = new Random(System.currentTimeMillis());
+
+    private Collection<Callable<Integer>> workloadTasks;
+
+    public PutWithDeltaTask(String id) {
+      this.id = id;
+    }
+
+    @Override
+    public void setUp(BenchmarkConfiguration cfg) throws Exception {
+      super.setUp(cfg);
+      this.workloadTasks = newSessionWorkloadTasks();
+      ClientCache cache = ClientCacheFactory.getAnyInstance();
+      cache.setCopyOnRead(true);
+      region = cache.getRegion("region");
+      save(Session.create(id));
+      this.sessionId.set(id);
+    }
+
+    @Override
+    public boolean test(Map<Object, Object> ctx) throws Exception {
+      int sessionAttributeCount = runSessionWorkload();
+
+      assert sessionAttributeCount == this.existingSessionAttributeNames.size();
+      Session session = findById(this.sessionId.get());
+
+      return true;
+    }
+
+    private Session findById(String id) {
+      return Optional.ofNullable(this.region.get(id))
+          .map(Session::commit)
+          .map(Session::touch)
+          .orElseThrow(() -> new IllegalStateException(
+              String.format("No Session with ID [%s] was found", id)));
+    }
+
+    private Session save(Session session) {
+      if (session != null && session.hasDelta()) {
+        this.region.put(session.getId(), session);
+        session.commit();
+      }
+
+      return session;
+    }
+
+    private int runSessionWorkload() {
+      return workloadTasks.stream().mapToInt(integerCallable -> {
+        try {
+          return integerCallable.call();
+        } catch (Exception e) {
+          e.printStackTrace();
+        }
+        return 0;
+      }).sum();
+    }
+
+    private Collection<Callable<Integer>> newSessionWorkloadTasks() {
+
+      Collection<Callable<Integer>> sessionWorkloadTasks = new LinkedList<>();
+
+      sessionWorkloadTasks.add(newAddSessionAttributeTask());
+      sessionWorkloadTasks.add(newRemoveSessionAttributeTask());
+      sessionWorkloadTasks.add(newSessionReaderTask());
+      return sessionWorkloadTasks;
+    }
+
+    private Callable<Integer> newAddSessionAttributeTask() {
+      return () -> {
+        Session session = findById(this.sessionId.get());
+
+        String name = UUID.randomUUID().toString();
```

Comment:
`UUID.randomUUID()` utilizes secure random to generate an array of bytes, which is not only more computationally more complex it also suffered similar contention issues that `Random` has. Also keeping in mind that those bytes are transient allocations and have impact on the GC and benchmark measurements.

If UUID is desired then you may want to construct UUIDs byte fetching 2 longs from `ThreadLocalRandom`.